src/main/java/core/basesyntax/servise/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.servise;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.servise.operation.OperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>(); // Хранилище фруктов
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+
+            handler.apply(fruit, quantity, storage);
+        }
+    }
+
+    public Map<String, Integer> getStorage() {
+        return storage;
+    }
```

Comment:
This method exposes the internal storage directly. It would be better to return a copy of the storage map to prevent external modifications to the internal state of the class.